zipline/pipeline/loaders/blaze/earnings.py

```diff
@@ -69,75 +59,21 @@ class BlazeEarningsCalendarLoader(PipelineLoader):
         ANNOUNCEMENT_FIELD_NAME,
     })
 
-    @preprocess(data_query_tz=optionally(ensure_timezone))
     def __init__(self,
                  expr,
                  resources=None,
                  odo_kwargs=None,
                  data_query_time=None,
                  data_query_tz=None,
-                 dataset=EarningsCalendar):
-        dshape = expr.dshape
-
-        if not istabular(dshape):
-            raise ValueError(
-                'expression dshape must be tabular, got: %s' % dshape,
-            )
-
-        expected_fields = self._expected_fields
-        self._expr = bind_expression_to_resources(
-            expr[list(expected_fields)],
-            resources,
-        )
-        self._odo_kwargs = odo_kwargs if odo_kwargs is not None else {}
-        self._dataset = dataset
-        check_data_query_args(data_query_time, data_query_tz)
-        self._data_query_time = data_query_time
-        self._data_query_tz = data_query_tz
-
-    def load_adjusted_array(self, columns, dates, assets, mask):
-        data_query_time = self._data_query_time
-        data_query_tz = self._data_query_tz
-        lower_dt, upper_dt = normalize_data_query_bounds(
-            dates[0],
-            dates[-1],
-            data_query_time,
-            data_query_tz,
-        )
-
-        raw = ffill_query_in_range(
-            self._expr,
-            lower_dt,
-            upper_dt,
-            self._odo_kwargs,
-        )
-        sids = raw.loc[:, SID_FIELD_NAME]
-        raw.drop(
-            sids[~sids.isin(assets)].index,
-            inplace=True
-        )
-        if data_query_time is not None:
-            normalize_timestamp_to_query_time(
-                raw,
-                data_query_time,
-                data_query_tz,
-                inplace=True,
-                ts_field=TS_FIELD_NAME,
-            )
-
-        gb = raw.groupby(SID_FIELD_NAME)
-
-        def mkseries(idx, raw_loc=raw.loc):
-            vs = raw_loc[
-                idx, [TS_FIELD_NAME, ANNOUNCEMENT_FIELD_NAME]
-            ].values
-            return pd.Series(
-                index=pd.DatetimeIndex(vs[:, 0]),
-                data=vs[:, 1],
-            )
-
-        return EarningsCalendarLoader(
-            dates,
-            valmap(mkseries, gb.groups),
-            dataset=self._dataset,
-        ).load_adjusted_array(columns, dates, assets, mask)
+                 dataset=EarningsCalendar,
+                 **kwargs):
+        super(
+            BlazeEarningsLoader, self
+        ).__init__(expr, dataset=dataset,
+                   resources=resources, odo_kwargs=odo_kwargs,
+                   data_query_time=data_query_time,
+                   data_query_tz=data_query_tz, **kwargs)
+
+    @property
+    def concrete_loader(self):
```

Comment:
Same as for `expected_cols` - I think it's a more elegant way of enforcing that subclasses implement this.